receiver/k8seventsreceiver/receiver_test.go

```diff
@@ -0,0 +1,170 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8seventsreceiver
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestNewReceiver(t *testing.T) {
+	rCfg := createDefaultConfig().(*Config)
+	client := fake.NewSimpleClientset()
+	r, err := newReceiver(
+		componenttest.NewNopReceiverCreateSettings(),
+		rCfg,
+		consumertest.NewNop(),
+		client,
+	)
+
+	require.NoError(t, err)
+	require.NotNil(t, r)
+	require.NoError(t, r.Start(context.Background(), componenttest.NewNopHost()))
+	r.Shutdown(context.Background())
+
+	rCfg.Namespaces = []string{"test", "another_test"}
+	r1, err := newReceiver(
+		componenttest.NewNopReceiverCreateSettings(),
+		rCfg,
+		consumertest.NewNop(),
+		client,
+	)
+
+	require.NoError(t, err)
+	require.NotNil(t, r1)
+	require.NoError(t, r1.Start(context.Background(), componenttest.NewNopHost()))
+	r1.Shutdown(context.Background())
```

Comment:
```suggestion
	require.NoError(t, r1.Shutdown(context.Background()))
```

Reply:
Actually, these last assertions should be `assert` instead of `require`

Reply:
And the same comment applies to other tests.

Reply:
I have made the change as per the commit suggestion.
AFAIK, the `require` package is very similar to `assert`, except `require` terminates test if the assertion is not successful so I would like to understand the reasoning behind changing the last assert to use `assert` instead of `require`.

Reply:
In a test ("given I have X, when I do Y, then Z happens"), you have typically three phases: setting up ("given"), running the test ("when"), and checking the results ("then"). Assertions are part of the last step. Anything that you need before that, is a requirement for the test.

So, use `require` for "given" and "when" phases, with `assert` for the "then" part.

Reply:
Thanks for the explanation, I have made the changes accordingly.